pythonfmu/pythonfmu-export/src/pythonfmu/PySlaveInstance.cpp

```diff
@@ -65,33 +143,17 @@ PySlaveInstance::PySlaveInstance(std::string instanceName, std::string resources
             handle_py_exception("[ctor] PyObject_GetAttrString", gilState);
         }
         int success = PyList_Insert(sys_path, 0, PyUnicode_FromString(resources_.c_str()));
+        PyList_Append(sys_path, PyUnicode_DecodeFSDefault(resources_.c_str()));
         Py_DECREF(sys_path);
         if (success != 0) {
             handle_py_exception("[ctor] PyList_Insert", gilState);
         }
 
         std::string moduleName = getline(resources_ + "/slavemodule.txt");
-        PyObject* pModule = PyImport_ImportModule(moduleName.c_str());
-        if (pModule == nullptr) {
-            handle_py_exception("[ctor] PyImport_ImportModule", gilState);
-        }
-
-        std::string className = findClassName(resources_ + "/" + moduleName + ".py");
-        if (className.empty()) {
-            cleanPyObject();
-            throw cppfmu::FatalError("Unable to find class extending Fmi2Slave!");
-        }
 
-        PyObject* pyClassName = Py_BuildValue("s", className.c_str());
-        if (pyClassName == nullptr) {
-            handle_py_exception("[ctor] Py_BuildValue", gilState);
-        }
-
-        pClass_ = PyObject_GetAttr(pModule, pyClassName);
-        Py_DECREF(pyClassName);
-        Py_DECREF(pModule);
+        pClass_ = findClass(resources_, moduleName);
         if (pClass_ == nullptr) {
-            handle_py_exception("[ctor] PyObject_GetAttr", gilState);
+            handle_py_exception("[ctor] Py_BuildValue", gilState);
```

Comment:
Should perhaps be handled on call site. This way it's confusing

Reply:
I don't understand what you mean by that, could you explain in more detail?

Reply:
I mean that `Py_BuildValue` is not called in this scope, but rather on line 107. Seeing this code at this point is confusing as context is missing. Changing the error to `handle_py_exception("[ctor] findClass", gilState);` might improve it (rather than moving to the function that provides the `nullptr`).

Reply:
Solved! Thank you!